### PR TITLE
 [android] Move Speed Cam preference into Voice Instruction preference menu

### PR DESCRIPTION
**Pull request by gpesquero**
Current "Speed Cameras" settings screen in Android shows a list of possible options ("Auto/Always/Never")...

android_1

...but it does not show a description of each option, as displayed in iOS:

ios

This PR adds a layout in the "Speed Cam" list preference in Android to show the same description as in iOS:

android_2

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Why not doing like in IOS and move this setting in TTS settings?

**Comment by biodranik** [comment]
> Why not doing like in IOS and move this setting in TTS settings?

That's a valid comment. It will require some work, but logically all voice-related settings should be in the TTS section. Otherwise, users may expect them to work even without setting up TTS properly. @gpesquero would it be hard to rewrite the current UI and move camera voice settings into the TTS dialog?

**Comment by gpesquero** [comment]
> @gpesquero would it be hard to rewrite the current UI and move camera voice settings into the TTS dialog?

Let me have a look at it and get back to you with my findings...

**Comment by gpesquero** [comment]
I've checked that it's quite direct to move the "Speed Cam" settings to the "Voice Instruction" settings menu, and there's also room for the description of the different speed cam options:

android_3

If you don't have any objection, I can move forward with this change under this same PR. Speed cam options shall be enabled/disabled based on availability/enabling of voice instructions, as it's done in iOS.

**Comment by gpesquero** [comment]
PR modified so it implements the move of Speed Cam preference into Voice Instruction preference menu. Speed Cam preference is enabled/disabled based on TTS availability/enabling.

It also displays the description of the Speed Cam options. Final layout matches the last screenshot uploaded yesterday.

I've also moved all the code of the initialization of all TTS-related preferences (Language selection, voice test, lang info & speed cams) into a single function `initTtsPrefScreen()`.

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14
Open TTS settings take more time, did you have any idea why?

**Comment by gpesquero** [comment]
Hi @Jean-BaptisteC !! I've made some measurements of the startup time of the TTS settings screen before and after this modification, and I've detected no relevant increase of the time. I have around 9 languages installed in the Android phone.

Do you have some figures of this time increase in your environment?

**Comment by Jean-BaptisteC** [comment]
I guess loading is a little less speed in debug.
I have retry with beta, loading time is same as latest version.

**Comment by Jean-BaptisteC** [comment]
@biodranik PTAL

**Comment by biodranik** [comment]
The current implementation can be improved. Why do we need short terms like "Auto" if users expect to see an explanation anyway? Let's remove short terms and always use longer strings instead.

**Comment by zyphlar** [comment]
While we're at it I'd love the GUI to say what the warning beep means. It's not identical to the speed camera announcement since it plays at a different time, so what does it mean and can it be disabled? The first few times I heard it I was scared I was doing something wrong, but didn't know what.

**Comment by gpesquero** [comment]
> The current implementation can be improved. Why do we need short terms like "Auto" if users expect to see an explanation anyway? Let's remove short terms and always use longer strings instead.

Removed the Auto/Always/Never short terms, using the long strings instead.

**Comment by gpesquero** [comment]
> While we're at it I'd love the GUI to say what the warning beep means. It's not identical to the speed camera announcement since it plays at a different time, so what does it mean and can it be disabled? The first few times I heard it I was scared I was doing something wrong, but didn't know what.

@zyphlar: I don't fully understand what exactly you're suggesting here... Just an information message close to the Speed Cam settings? Which message exactly? I'd recommend you to open a new issue, with the detailed explanation of your proposal...

File: data/strings/strings.txt [hunk 1]
```diff
@@ -22858,6 +22858,141 @@
...
+    da = Advar om fartkameraer, hvis der er en risiko for, at hastighedsgrænsen overskrides
+    de = Vor Blitzern warnen, wenn die Gefahr einer Geschwindigkeitsüberschreitung besteht
+    el = Να ειδοποιούμαι για κάμερες αν υπάρχει κίνδυνος να υπερβώ το όριο ταχύτητας
+    es = Advierte sobre las cámaras de velocidad si existe el riesgo de exceder el límite de velocidad
+    es-MX = Aviso de las cámaras de velocidad si existe el riesgo de exceder el límite de velocidad
+    et = Kiiruskaamerate hoiatused, kui on oht ületada piirkiirust
+    eu = Abiadura-kamerei buruz ohartarazten du abiadura muga gainditzeko arriskua badago
+    fa = در صورتی که خطر تجاوز از سرعت مجاز وجود داشت، نسبت به دوربین‌های کنترل سرعت هشدار دهد
+    fi = Varoita nopeuskameroista, jos tuntuu siltä, että nopeusrajan ylitys on mahdollista
+    fr = Alerte sur les radars en cas de risque de dépassement de la limite de vitesse
```
Comment (Jean-BaptisteC) [inline data/strings/strings.txt:22880]:
```suggestion
    fr = Alerte sonore en cas de risque de dépassement de la limite de vitesse
```

File: data/strings/strings.txt [hunk 2]
```diff
@@ -22858,6 +22858,141 @@
...
+    da = Advar altid om fartkameraer
+    de = Immer vor Blitzern warnen
+    el = Πάντα να ειδοποιούμαι για κάμερες
+    es = Siempre alerta sobre las cámaras
+    es-MX = Siempre dar aviso de las cámaras de velocidad
+    et = Hoiata alati kiiruskaamerate eest
+    eu = Beti abisatu kamerei buruz
+    fa = همیشه درباره دوربین‌های کنترل سرعت هشدار دهد
+    fi = Varoita kameroista aina
+    fr = Toujours avertir des radars
```
Comment (Jean-BaptisteC) [inline data/strings/strings.txt:22925]:
```suggestion
    fr = Toujours avertir
```

File: data/strings/strings.txt [hunk 3]
```diff
@@ -22858,6 +22858,141 @@
...
+    da = Advar aldrig om fartkameraer
+    de = Niemals vor Blitzern warnen
+    el = Ποτέ να μην ειδοποιούμαι για κάμερες
+    es = Nunca advierte acerca de las cámaras
+    es-MX = Nunca dar aviso de las cámaras de velocidad
+    et = Ära kunagi hoiata kiiruskaamerate eest
+    eu = Inoiz ez abisatu kamerei buruz
+    fa = هرگز درباره دوربین‌های کنترل سرعت هشدار ندهد
+    fi = Älä koskaan varoita kameroista
+    fr = Jamais avertir à propos des radars
```
Comment (Jean-BaptisteC) [inline data/strings/strings.txt:22970]:
```suggestion
    fr = Jamais avertir
```

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14

File: data/strings/strings.txt (hunk 1 above)
Comment (gpesquero) [reply data/strings/strings.txt:22880]: Updated!!

File: data/strings/strings.txt (hunk 2 above)
Comment (gpesquero) [reply data/strings/strings.txt:22925]: Updated!!

File: data/strings/strings.txt (hunk 3 above)
Comment (gpesquero) [reply data/strings/strings.txt:22970]: Updated!!

File: data/strings/strings.txt [hunk 4]
```diff
@@ -22858,6 +22858,141 @@
     zh-Hans = 加载地图
     zh-Hant = 載入地圖
 
+  [pref_tts_speedcams_auto]
+    tags = android
+    en = Warn about speedcams if there is a risk of exceeding the speed limit
```
Comment (biodranik) [inline data/strings/strings.txt:22863]: Thanks! If I understood correctly, the UI now looks like this:

[Speed cameras]
[Warn about speedcams if there is a risk of exceeding the speed limit]

As there is already [Speed cameras] title, it's better to simplify all strings by removing "about speedcams":

[Speed cameras]
[Warn if there is a risk of exceeding the speed limit]

Comment (gpesquero) [reply data/strings/strings.txt:22863]: This is how the UI looks like right now...
and_01


And this is after the removal of "about speedcams" string:
and_02

Comment (gpesquero) [reply data/strings/strings.txt:22863]: The main issue here is going to be the removal of the "about speedcams" in the strings of the different languages...

I'm going to try to do it myself.. It should not be hard for most of the languages, but for some of them (specially asian or arabic), my changes could be a really big mess...

We can request the help of the translation community here...

Comment (gpesquero) [reply data/strings/strings.txt]: I've updated the commit with the removal of "about speedcams" in the strings for all languages.

For all languages, except these 4 (which  I find quite hard to modify without knowledge...):
- ja (Japanese)
- th (Thai)
- zh-Hans (Chinese simplified)
- zh-Hant (Chinese traditional)

File: data/strings/strings.txt [hunk 5]
```diff
@@ -22904,6 +22904,141 @@
     zh-Hans = 加载地图
     zh-Hant = 載入地圖
 
+  [pref_tts_speedcams_auto]
+    tags = android
```
Comment (biodranik) [inline data/strings/strings.txt:22908]: @gpesquero can you please add a `comment =` tag with a bit more details about the context for translators?

File: data/strings/strings.txt [hunk 6]
```diff
@@ -22904,6 +22904,141 @@
...
+    en = Warn if there is a risk of exceeding the speed limit
+    af = Waarsku indien daar ’n risiko is dat die grens oorskry word
+    ar = حذرني  إذا هناك احتمالية تجاوز الحد الأقصى للسرعة
```
Comment (biodranik) [inline data/strings/strings.txt:22911]: @organicmaps/translations please check if the strings below are correctly translated in your language. Thanks in advance!

Comment (vgrigoriu) [reply data/strings/strings.txt:22911]: Romanian looks good.

File: data/strings/strings.txt [hunk 7]
```diff
@@ -22904,6 +22904,141 @@
...
+    id = Peringatkan saat terdapat risiko melebihi batas kecepatan
+    it = Avvisare in caso di rischio eccesso velocità
+    ja = 速度制限超過の危険性がある場合には、自動速度違反取締装置について警告します
+    ko = 속도 제한 초과 위험이 있을때 경고하기
+    lt = įspėti jei yra greičio viršijimo rizika
+    mr = वेगमर्यादा ओलांडण्याचा धोका असल्यास चेतावणी द्या
+    nb = Advarsel hvis det er fare for å overskride fartsgrensen
+    nl = Waarschuw als er een risico bestaat dat de snelheidslimiet wordt overschreden
+    pl = Ostrzegaj jeśli istnieje ryzyko przekroczenia ograniczenia prędkości
+    pt = Avisa se houver risco de exceder o limite de velocidade
```
Comment (matheusgomesms) [inline data/strings/strings.txt:22938]:
```suggestion
    pt = Avisar se houver risco de exceder o limite de velocidade
```

File: data/strings/strings.txt [hunk 8]
```diff
@@ -22904,6 +22904,141 @@
...
+    id = Selalu peringatkan
+    it = Avvisare sempre
+    ja = 自動速度違反取締装置について常に警告します
+    ko = 항상 경고하기
+    lt = Visada įspėti
+    mr = नेहमी चेतावणी द्या
+    nb = Advar alltid
+    nl = Waarschuw altijd
+    pl = Zawsze ostrzegaj
+    pt = Avisa sempre
```
Comment (matheusgomesms) [inline data/strings/strings.txt:22983]:
```suggestion
    pt = Sempre avisar
```

File: data/strings/strings.txt (hunk 7 above)
Comment (gpesquero) [reply data/strings/strings.txt:22938]: Thanks @matheusgomesms. Commit amended!

File: data/strings/strings.txt (hunk 8 above)
Comment (gpesquero) [reply data/strings/strings.txt:22983]: Thanks @matheusgomesms. Commit amended!

File: data/strings/strings.txt (hunk 5 above)
Comment (gpesquero) [reply data/strings/strings.txt:22908]: > @gpesquero can you please add a `comment =` tag with a bit more details about the context for translators?

Done!

File: android/app/src/main/res/values-ca/strings.xml [hunk 9]
```diff
@@ -629,6 +629,12 @@
 	<string name="place_description_title">Descripció del lloc</string>
 	<!-- this text will be shown in application notification preferences opposite checkbox which enable/disable downloader notifications. Devices on Android 8+ are affected. -->
 	<string name="notification_channel_downloader">Descàrrega de mapes</string>
+	<!-- Speed camera settings menu option - Warn (about speedcams) if risk of exceeding speed limit -->
+	<string name="pref_tts_speedcams_auto">Avisa si hi ha risc d\'excedir el límit de velocitat</string>
+	<!-- Speed camera settings menu option - Always warn (about speedcams) -->
+	<string name="pref_tts_speedcams_always">Sempre avisa</string>
+	<!-- Speed camera settings menu option - Never warn (about speedcams) -->
+	<string name="pref_tts_speedcams_never">No avisa mai</string>
```
Comment (fitojb) [inline android/app/src/main/res/values-ca/strings.xml:637]: No n'avisis mai

File: android/app/src/main/res/values-es/strings.xml [hunk 10]
```diff
@@ -640,6 +640,12 @@
 	<string name="place_description_title">Descripción del lugar</string>
 	<!-- this text will be shown in application notification preferences opposite checkbox which enable/disable downloader notifications. Devices on Android 8+ are affected. -->
 	<string name="notification_channel_downloader">Descargar mapas</string>
+	<!-- Speed camera settings menu option - Warn (about speedcams) if risk of exceeding speed limit -->
+	<string name="pref_tts_speedcams_auto">Alertar si existe el riesgo de exceder el límite de velocidad</string>
+	<!-- Speed camera settings menu option - Always warn (about speedcams) -->
+	<string name="pref_tts_speedcams_always">Alertar siempre</string>
+	<!-- Speed camera settings menu option - Never warn (about speedcams) -->
+	<string name="pref_tts_speedcams_never">No advertir nunca</string>
```
Comment (fitojb) [inline android/app/src/main/res/values-es/strings.xml:648]: No alertar nunca

Comment (gpesquero) [reply android/app/src/main/res/values-es/strings.xml]: Commit amended!!

File: android/app/src/main/res/values-ca/strings.xml (hunk 9 above)
Comment (gpesquero) [reply android/app/src/main/res/values-ca/strings.xml]: Commit amended!!

**Review by biodranik — APPROVED** [review]
Thank you!

**Pull request merged**